chemistry/structure.py

```diff
@@ -919,11 +919,17 @@ def update_dihedral_exclusions(self):
         turns it to `True` if the two end atoms are in the bond or angle
         partners arrays
         """
+        set14 = set()
         for dihedral in self.dihedrals:
-            if not dihedral.ignore_end: continue
+            if dihedral.ignore_end : continue
             if (dihedral.atom1 in dihedral.atom4.bond_partners or
                 dihedral.atom1 in dihedral.atom4.angle_partners):
                 dihedral.ignore_end = True
+            elif (dihedral.atom1.idx, dihedral.atom4.idx) in set14:
+                # Avoid double counting of 1-4 in a six-membered ring
+                dihedral.ignore_end = True
+            else:
+                set14.add((dihedral.atom1.idx, dihedral.atom4.idx))
```

Comment:
Don't you need to add `(dihedral.atom4.idx, dihedral.atom1.idx)` as well to make sure that line 928 doesn't miss that permutation?